### ISSUE TITLE
`file_put_contents()` error in `FileReferenceCacheProvider`

### ISSUE THREAD

**Issue by AdamMarton** [psalm/psalm-vscode-plugin#6]
After [figuring out](https://github.com/psalm/psalm-vscode-plugin/issues/3#issuecomment-510483292) how to fire up Psalm VSCode plugin, I'm getting the following error:

```
PHP Warning:  file_put_contents(C:\Users\ADAM_M~1\AppData\Local\Temp/psalm\4bd298dd2c6ac22b6a075c5730e13891ec2b484f-s\config):\\
failed to open stream: No such file or directory in\\
C:\Users\Adam_Marton\AppData\Roaming\Composer\vendor\vimeo\psalm\src\Psalm\Internal\Provider\FileReferenceCacheProvider.php on line 650
```

**Comment by muglug** [comment]
That’s a bug in Psalm (but should be a one-time innocuous thing). Should be fixed in the next release.

**Comment by AdamMarton** [comment]
Hi @muglug , thanks for the quick turnaround.
Can you say the same about undefined classes? Because that's the very next error in devtools.

**Comment by muglug** [comment]
What classes are complained about?

**Comment by AdamMarton** [comment]
This is a WordPress project with its own plugins which are referencing each other. This is fine in WordPress, but it isn't help during static analysis.

DevTools shows a 37 nodes backtrace, like this:

```
Error: Class 'acf_field' not found in ...
rejected promise not handled within 1 second: Error: Error: Class 'acf_field' not found in ...
...
```

**Comment by muglug** [comment]
Does Psalm run on the codebase on its own?  (not inside VS Code)

**Comment by AdamMarton** [comment]
I've almost told you it is, but fortunately ran it before sending my comment.

No, it doesn't (and it used to work with this code).

**Comment by muglug** [comment]
The original issue you mention is now fixed in Psalm master

**Comment by muglug** [comment]
Please first check that the new issue isn't fixed in Psalm's latest master, then ticket the Psalm failure at https://github.com/vimeo/psalm (and if you can identify when Psalm started to fail, all the better).
